--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,21 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go-tree-sitter",
+    srcs = glob(
+        include = [
+            "*.go",
+            "*.c",
+            "*.h",
+        ],
+        exclude = [
+            "*_test.go",
+            "tree_sitter.go",
+        ],
+    ),
+    cdeps = ["@tree-sitter"],
+    cgo = True,
+    importpath = "github.com/tree-sitter/go-tree-sitter",
+    visibility = ["//visibility:public"],
+    deps = ["@com_github_mattn_go_pointer//:go-pointer"],
+)
